signac/common/config.py

```diff
@@ -40,10 +40,7 @@ def _locate_config_dir(search_path):
     while True:
         if os.path.isfile(_get_project_config_fn(root)):
             return root
-        # TODO: Could use the walrus operator here when we completely drop
-        # Python 3.7 support if we like the operator.
-        up = os.path.dirname(root)
-        if up == root:
+        if (up := os.path.dirname(root)) == root:
```

Comment:
are `root`s remaining here without being changed into search_path?

Reply:
Good idea. I refactored this in ba109a2ba371079ac36eaf21b3fb7c189abe5090.